packages/components/src/Card/Card.mdx

```diff
@@ -112,3 +113,62 @@ Card titles should be sentence-cased.
     </Content>
   </Card>
 </Playground>
+
+### Header with button
+
+<Playground>
+  <Card
+    header={{
+      title: "Banana varieties",
+      action: {
+        label: "Buy",
+        size: "small",
+        type: "primary",
+        icon: "add",
+        onClick: () => {
+          alert("Buy Bananas!");
+        },
+      },
+    }}
+  >
+    <Content>
+      <Heading level={4}>Burro</Heading>
+      <Text>
+        Squatty and slightly square at the edges, these bananas are slightly
+        tangy and lemony. When ripe, the fruit is soft and the skin is yellow
+        with black spots. Also called Horse, Hog, or Orinoco bananas.
+      </Text>
+      <Heading level={4}>Cavendish</Heading>
+      <Text>
+        This is the familiar yellow banana sold in United States’ supermarkets.
+        There are also other sizes of Cavendish, including Dwarf and Giant,
+        though they are difficult to find.
+      </Text>
+    </Content>
+  </Card>
+</Playground>
+
+### Header with custom component
+
+<Playground>
+  <Card
+    header={{
+      customHeader: <Banner type="notice">Bananas are on sale!</Banner>,
+    }}
```

Comment:
We should skip the object and go straight to

```ts
    header={<Banner type="notice">Bananas are on sale!</Banner>}
```

if it's a custom component. This also removes the need for an XOR since TS (and JS) can differentiate between an object or a function/component `HeaderActionProps | ReactNode`

Also, we shouldn't use banner unless the designer wants that to happen. Is that the intended use for it?

Reply:
➕ to a more realistic example. I can see something like a `Heading level={4}` and a `subtle` Button with a cross Icon and no grey background being something that gets used for cases like inline messages.

Reply:
@darryltec  not a specific design we intend to have, I don't have a concrete example using custom component other a banner or adding multiple action buttons. @chris-at-jobber does it work with this sample 
![image](https://user-images.githubusercontent.com/94410044/218193769-757d887e-b856-4ada-a281-a99d412f3a35.png)

Reply:
That looks good, if we could make the button `tertiary` instead so there's no border that would look ideal 👌